client/components/data/query-post-counts/index.jsx

```diff
@@ -1,58 +1,35 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestPostCounts } from 'state/posts/counts/actions';
 import { isRequestingPostCounts } from 'state/posts/counts/selectors';
 
-class QueryPostCounts extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId && this.props.type === nextProps.type ) {
-			return;
-		}
-
-		this.request( nextProps );
-	}
+export default function QuerySiteDomains( { siteId, type } ) {
+	const requesting = useSelector( state => isRequestingPostCounts( state, siteId, type ) );
+	const dispatch = useDispatch();
+	const previousId = useRef( undefined );
+	const previousType = useRef( undefined );
 
-	request( props ) {
-		if ( props.requesting ) {
-			return;
+	useEffect( () => {
+		if ( ! requesting && siteId !== previousId.current && type !== previousType.current ) {
+			requestPostCounts( siteId, type )( dispatch );
 		}
 
-		props.requestPostCounts( props.siteId, props.type );
-	}
+		previousId.current = siteId;
+		previousType.current = type;
+	}, [ dispatch, requesting, siteId, type ] );
```

Comment:
Again, it would be great if we could avoid the refs and write just:
```js
useEffect( () => {
  request( siteId, type );
}, [ siteId, type ] );
```
The `useEffect` hook itself remembers the previous values and compares against them. And we are kind of duplicating that check.

I know it's not that straightforward because we want to avoid issuing the request again when the `requesting` flag resets from `true` to `false`.

One solution is to omit `requesting` from the list of dependencies.

Reply:
Switched all over the PR 👍